### DISCUSSION TITLE
How can I optimize hyperparameters automatically?

### DISCUSSION THREAD

**Discussion by MarleneBs** [awslabs/gluonts#2419] [Q&A] [answered]
How can I automatically optimize hyperparameters if I want to have the data in the following form as a training and test split:

```
to_pandasdataset = lambda data: PandasDataset(
    data,
    feat_dynamic_real=["dynamic_real_1"]
)
train = to_pandasdataset([ts.iloc[:-prediction_length, :] for ts in multiple_ts])
test = to_pandasdataset(multiple_ts)
```

And I also want to give a certain number of feat_dynamic_real with

**Comment by jaheba** [comment, marked as answer]
Hi,

there is a tutorial explaining how to do HPO using Optuna: https://ts.gluon.ai/stable/tutorials/advanced_topics/hp_tuning_with_optuna.html

Also, there is SyneTune: https://github.com/awslabs/syne-tune/blob/main/examples/launch_tuning_gluonts.py
